conjure-java-core/src/main/java/com/palantir/conjure/java/services/dialogue/DefaultStaticFactoryMethodGenerator.java

```diff
@@ -221,14 +294,27 @@ private MethodSpec clientImpl(EndpointDefinition def) {
                 Names.endpointChannel(def),
                 REQUEST,
                 def.getReturns()
-                        .filter(type -> returnTypes.isBinaryOrOptionalBinary(returnTypes.baseType(type)))
+                        .filter(type -> !options.generateErrorParameterFormatRespectingDialogueInterfaces()
+                                && returnTypes.isBinaryOrOptionalBinary(returnTypes.baseType(type)))
                         .map(type -> StaticFactoryMethodGenerator.RUNTIME
                                 + (returnTypes.isOptionalBinary(returnTypes.baseType(type))
                                         ? ".bodySerDe().optionalInputStreamDeserializer()"
                                         : ".bodySerDe().inputStreamDeserializer()"))
                         .orElseGet(() -> def.getEndpointName().get() + "Deserializer"));
 
         methodBuilder.addCode(request);
+        if (options.generateErrorParameterFormatRespectingDialogueInterfaces()) {
+            // If the BodySerDe.errorParameterFormat field is set, add the header
+            methodBuilder.addCode(CodeBlock.builder()
+                    .beginControlFlow("if ($L.bodySerDe().errorParameterFormat().isPresent())", RUNTIME)
+                    .add(
+                            "$L.putHeaderParams($S," + " $L.bodySerDe().errorParameterFormat().get().toString());",
+                            REQUEST,
+                            "Accept-Conjure-Error-Parameter-Format", // TODO(pm): make this public.
```

Comment:
Specifically flagging your TODO here (I imagine this is a constant in dialogue already? not sure which PR is adding it and it doesn't seem to be on develop)